src/elements/fields/DateSelectorField/index.tsx

```diff
@@ -201,20 +200,24 @@ function DateSelectorField({
         <DatePicker
           id={element.servar.key}
           selected={internalDate}
-          preventOpenOnFocus
           autoComplete='off'
           onCalendarOpen={handleCalendarOpen}
           onCalendarClose={handleCalendarClose}
           onSelect={onDateChange} // when day is clicked
-          onChange={onDateChange} // only when value has changed
+          onChange={setInternalDate} // only when value has changed
           onFocus={(e: any) => {
             if (isTouchDevice()) {
               // hide keyboard on mobile focus
               e.target.readOnly = true;
             }
+            // select all text on focus
+            e.target.select();
             setFocused(true);
           }}
-          onBlur={() => setFocused(false)}
+          onBlur={() => {
```

Comment:
i believe onDateChange should also be triggered if enter is typed as well now

Reply:
also, what if i type in a date and immediately hit the submit button on the step? are we guaranteed this blur event will fire first and propagate state change before the submit logic validates?